filebeat/filebeat.spec

```diff
@@ -0,0 +1 @@
+{"BinaryPath":"filebeat","Args":["-e"],"Configurable":"grpc"}
```

Comment:
Can we move that to YAML spec files?

Reply:
do you mean classic filebeat.yml?

Reply:
I am talking about the files in x-pack/agent/spec.

Reply:
x-pack/agent/spec are transformation rules. this file should describe specific binary. 
if we stick to agent.version == beat.version it is fine to have it there. but we probably still need to support `{program}.spec` for not yet known programs. e.g running endpoint or community beat eventually, so agent knows if specific binary supports grpc protocol or if there are some arguments needed to start binary correctly.
i'm not sure what is a correct way of handling this. maybe lookup local spec + fallback to `{program}.spec` with local having priority?

Reply:
For the endpoint case, I think we would just have a specific spec for them. Concerning the community beats its a complete other madness and open the door to all sort of problems. We would need a way to add signatures and support the signing of community beats or expose that. :)

For the POC I think we can keep that way but let's create a followup issue to refactor that this sound like a P3 for the priority.

Reply:
i created a new issue here: https://github.com/elastic/beats/issues/14985